Automattic/wp-calypso
# Gutenboarding: Add separate entrypoint

#### Changes proposed in this Pull Request

Branched from @jsnajdr's #37071 (thanks Jarda!). ~Requires #37137 to be merged first.~ ✅ 

> [The first] commit adds a `style-bare.scss` stylesheet with just the CSS resets and color schemes, without Calypso-opinionated styles.
>
> [The t]hird commit adds the actual entrypoint. [...] It renders the [...] `<Gutenboard />` component. There is no Redux at all in this entrypoint, it just bootstraps the user (so that `user.get()` returns the logged-in user) and then renders a React component. It also doesn't use the Calypso `<Layout>` component. Renders directly into the `wpcom` div.
>
> One thing that doesn't work is the loading placeholder that server sends and browser displays while JS is loading. It's unstyled, because its styles are part of the Calypso-opinionated `main` stylesheet.

One major addition on top of Jarda's code is moving the Gutenboarding code from `client/gutenboarding/` into `client/landing/gutenboarding/`, to bring it in line with other separate entrypoints, and consequently removing the 'hard-wired' section definition from `sections.js` (since those are sections that are by definition for the main entrypoint).

#### Testing instructions

Visit `calypso.localhost:3000/gutenboarding`. Verify that it still works.

Note that some styling looks better (dropdown caret), while some other looks worse (fonts). This is due to the absence of some Calypso 'basic' styling, which was one of the reasons why we went for this approach -- to be able to start from a cleaner slate with regard to styling. We'll work on adding back the bits that we want.
